var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -159,6 +159,10 @@ class Petsc(Package, CudaPackage, ROCmPackage):
 
     # https://github.com/spack/spack/issues/37416
     conflicts("^rocprim@5.3.0:5.3.2", when="+rocm")
+    # petsc 3.20 has workaround for breaking change in hipsparseSpSV_solve api,
+    # but it seems to misdetect hipsparse@5.6.1 as 5.6.0, so the workaround
+    # only makes things worse
+    conflicts("^hipsparse@5.6", when="+rocm @3.20.0")
```

Comment:
```suggestion
    # hipsparse@5.6.0 changed hipsparseSpSV_solve() API, but reverted in 5.6.1.
    #  petsc has a fixes that works with 5.6.0, 5.7.0 - but not 5.6.1 - as its detected as 5.6.0 (instead of 5.6.1)
        conflicts("^hipsparse@5.6.1", when="+rocm")
```

Notes:
- petsc-3.19, 3.20 should work with 5.6.0
- petsc-3.20 should work with 5.7.0